### PR TITLE
Get /api

### PR DESCRIPTION
**Pull request by omar-baggio**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -34,3 +35,23 @@ describe("GET/api/topics", () => {
...
 });
+
+describe("GET/api", () => {
+  test("status: 200, should respond with a json file describing each endpoint", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toEqual(endpoints);
+      });
```
Comment (Jodiech) [inline __tests__/app.test.js:46]: You should be more specific about what form this JSON file should take. Maybe have a look at .toMatchObject( ) and expect.objectContaining( ) 👀

File: controller/api.controller.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const endpoints = require("../endpoints.json");
+
+exports.getApi = (req, res, next) => {
+  res.status(200).send(endpoints);
```
Comment (Jodiech) [inline controller/api.controller.js]: I'd consider sending this back destructured 👀

File: __tests__/app.test.js [hunk 3]
```diff
@@ -34,3 +35,23 @@ describe("GET/api/topics", () => {
...
+  });
+
+  test("404: responds with error message page not found", () => {
```
Comment (Jodiech) [inline __tests__/app.test.js:51]: You can do a test for a 404, but it shouldn't be in the same describe block for this ticket as it's not specific to this endpoint. Please separate this out 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 great work on this ticket! There are just a few tweaks to be made to your tests, then you're good to go! Please submit another PR once you have made these changes 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -34,3 +35,25 @@ describe("GET/api/topics", () => {
       });
   });
 });
+
+describe("GET/api", () => {
+  test("status: 200, should respond with a json file describing each endpoint", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body: { endpoints } }) => {
```
Comment (lewispricey) [inline __tests__/app.test.js:44]: Nice use of destructuring here to access the endpoints key

**Review by lewispricey — APPROVED** [review]
Great work on this ticket Omar, thanks for implementing the changes Jodie suggested.

For your api endpoint tests you could check that each of the endpoint keys ("GET /api" ect.) contains an object that has the 4 expected properties rarther than checking the endpoint keys themselves (mainly so you don't have to go back and update this test for every endpoint) but what you have gives us confidence that you are recieving the endpoint data back so this is absolutely fine. 💯 

Good to merge when you're ready 🚀

**Pull request merged**